packages/web/src/features/translation/TranslationView.tsx

```diff
@@ -1,42 +1,137 @@
-  const hebrew = bookId < 40;
+      const queryKey = ['verse-glosses', language, params.verseId];
+      const query =
+        queryClient.getQueryData<GetVerseGlossesResponseBody>(queryKey);
+      if (query) {
+        const glosses = query.data.slice();
+        const index = glosses.findIndex((g) => g.wordId === wordId);
+        glosses.splice(index, 1, { wordId, gloss });
+        queryClient.setQueryData<GetVerseGlossesResponseBody>(queryKey, {
+          data: glosses,
+        });
+      }
+    },
+  });
 
-  const { language } = useLayoutContext();
+  const loading =
+    !verseQuery.isSuccess ||
+    !referenceGlossesQuery.isSuccess ||
+    !targetGlossesQuery.isSuccess;
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  const verse = verseQuery.data.data;
+  const referenceGlosses = referenceGlossesQuery.data.data;
+  const targetGlosses = targetGlossesQuery.data.data;
+
+  const bookId = parseInt(verse.id.slice(0, 2));
+  const isHebrew = bookId < 40;
 
   return (
     <div className="px-4">
-      <ol className={`flex flex-wrap ${hebrew ? 'flex-row-reverse' : ''}`}>
-        {verse.data.words.map((word, i) => (
-          <li key={word.id} className="mx-2 mb-4">
-            <div className={`font-serif ${hebrew ? 'text-2xl' : 'text-lg'}`}>
-              {word.text}
-            </div>
-            <div className="">{referenceGlosses.data[i]?.gloss}</div>
-          </li>
-        ))}
+      <ol className={`flex flex-wrap ${isHebrew ? 'flex-row-reverse' : ''}`}>
+        {verse.words.map((word, i) => {
+          const targetGloss = targetGlosses[i]?.gloss;
+          const isSaving = glossRequests.some(
+            ({ wordId }) => wordId === word.id
+          );
+
+          return (
+            <li key={word.id} className="mx-2 mb-4 w-36">
+              <div
+                className={`font-serif mb-2 ${
+                  isHebrew ? 'text-2xl text-right' : 'text-lg'
+                }`}
+              >
+                {word.text}
+              </div>
+              <div className="mb-2">{referenceGlosses[i]?.gloss}</div>
+              <TextInput
+                className="w-full"
+                defaultValue={targetGloss}
+                aria-describedby={`word-help-${word.id}`}
+                onBlur={async (e) => {
+                  const newGloss = e.currentTarget.value;
+                  if (newGloss !== targetGloss) {
+                    glossMutation.mutate({
+                      wordId: word.id,
+                      gloss: newGloss,
+                    });
+                  }
+                }}
+              />
+              <InputHelpText id={`word-help-${word.id}`}>
+                {(() => {
+                  if (isSaving) {
+                    return (
+                      <>
+                        <Icon icon="arrows-rotate" className="mr-1" />
+                        Saving...
+                      </>
+                    );
+                  } else if (targetGloss) {
```

Comment:
Under the input, we show a saving indicator so that the user feels confident their data is persisted. This uses a self invoking function so that we can write if statements.